### PR TITLE
Use edge definitions in query paths

### PR DESCRIPTION
**Pull request by TimDiekmann**
## 🌟 What is the purpose of this PR?

To store reversed edges it's easier to store the edge kind in the query path than both edges. For types with the same right endpoint of an edge (currently entity types and entities), this also avoids duplication of these definitions.

## 🔗 Related links

- [Asana task](https://app.asana.com/0/1204000740778935/1204165139634983/f) _(internal)_

## 🚫 Blocked by

- #2202

## 🔍 What does this change?

- Use the edge definitions to identify query paths

## ⚠️ Known issues

With the current implementation, there is not much advantage to change the definitions of the query paths, however, with the two follow-up PRs, this will be more clear.

## 🐾 Next steps

- [Support reversed edges in structural ontology queries](https://app.asana.com/0/1204000740778935/1204165139634979/f) _(internal)_
- [Migrate the current implementation of looking up ontology type references in memory to instead be from the linking tables in DB](https://app.asana.com/0/1204000740778935/1204117847656666/f) _(internal)_

## 🛡 What tests cover this?

Basically every test, which uses structural query. Also a new unit test for a missing filter was added.

### CODE REVIEW

File: apps/hash-graph/lib/graph/src/knowledge/entity/query.rs [hunk 1]
```diff
@@ -108,99 +109,129 @@ pub enum EntityQueryPath<'p> {
...
     /// [`Entity`]: crate::knowledge::Entity
-    /// [`EntityMetadata`]: crate::knowledge::EntityMetadata
-    /// [`EntityType`]: type_system::EntityType
-    Type(EntityTypeQueryPath<'p>),
-    /// Represents an [`Entity`] linking to the [`Entity`].
+    EntityTypeEdge {
+        edge_kind: SharedEdgeKind,
+        path: EntityTypeQueryPath<'p>,
+    },
+    /// An edge between two [`Entities`][`Entity`] using a [`KnowledgeGraphEdgeKind`].
```
[the remark this answers is not in the document]
Comment (TimDiekmann) [reply apps/hash-graph/lib/graph/src/knowledge/entity/query.rs:137]: I didn't want to use "Entitys" but `Entities` as this is the correct plural. However, it should link to `Entity`. It's unfortunate but this was on purpose.

[the remark this answers is not in the document]
Comment (TimDiekmann) [reply apps/hash-graph/lib/graph/src/knowledge/entity/query.rs:137]: Yes, exactly 🙂

**Review by thehabbos007 — APPROVED** [review]
